GoMeet/src/test/java/com/google/sps/LocationDaoTest.java

```diff
@@ -0,0 +1,166 @@
+package test.java.com.google.sps;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.*;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import main.java.com.google.sps.servlets.LocationServlet;
+import main.java.com.google.sps.data.Location;
+import main.java.com.google.sps.data.LocationDao;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for LocationDao.java */
+@RunWith(JUnit4.class)
+public class LocationDaoTest {
+  private final long INIT_VOTE_COUNT = 1;
+  private final Location LOCATION_A = new Location("Cake Shop", 10.0, 15.0, "I like Cakes!", 1);
+  private final Location LOCATION_B = new Location("Fruit Shop", 22.0, 60.0, "", 1);
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private DatastoreService ds;
+  private LocationDao locationDao;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    ds = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Does this require a real connection to the datastore service or is this faked?

Reply:
It is faked. It is a local implementation of Datastore.

https://cloud.google.com/appengine/docs/standard/java/tools/localunittesting